elfpy/markets/hyperdrive/yieldspace_pricing_model_sol.py

```diff
@@ -0,0 +1,111 @@
+"""The Hyperdrive pricing model"""
+from __future__ import annotations
+
+from fixedpointmath import FixedPoint
+
+# Let the variable names be the same as their solidity counterpart so that it is easier to compare
+# the two.  We can make python wrappers that just call these methods that have better variable names
+# that conform to python standards.
+# pylint: disable=invalid-name
+# pylint: disable=too-many-arguments
+
+ONE_18 = FixedPoint("1")
+
+
+def modified_yield_space_constant(
+    c_div_mu: FixedPoint, mu: FixedPoint, z: FixedPoint, t: FixedPoint, y: FixedPoint
+) -> FixedPoint:
+    r"""
+    Helper function to derive invariant constant C for the YieldSpace.
+    This is meant to mirror the solidity.
+
+    Parameters
+    ----------
+    c_div_mu : FixedPoint
+        Normalized price of shares in terms of base.
+    mu : FixedPoint
+        Interest normalization factor for shares.
+    z : FixedPoint
+        Amount of share reserves in the pool.
+    t : FixedPoint
+        Amount of time elapsed since term start.
+    y : FixedPoint
+        Amount of bond reserves in the pool.
+
+    Returns
+    -------
+    FixedPoint
+        The modified YieldSpace constant C.
+    """
+    # calculate and return the modified YieldSpace constant using regular arithmetic operators
+    return c_div_mu * (mu * z) ** t + y**t
+
+
+def calculate_bonds_out_given_shares_in(
```

Comment:
this exists: https://github.com/delvtech/elf-simulations/blob/0341920289309f93f9ade98cb89db2c1477d3d7d/elfpy/markets/hyperdrive/yieldspace_pricing_model.py#L560